client/state/jetpack-settings/modules/actions.js

```diff
@@ -62,3 +71,44 @@ export const deactivateModule = ( siteId, moduleSlug ) => {
 			} );
 	};
 };
+
+/**
+ * Returns an action object used in signalling that the available modules
+ * in a Jetpack site were received.
+ *
+ * @param  {Number}   siteId    Site ID
+ * @param  {Object[]} modules Object of modules indexed by slug
+ * @return {Object}             Action object
+ */
+export function receiveJetpackModules( siteId, modules ) {
+	return {
+		type: JETPACK_MODULES_RECEIVE,
+		siteId,
+		modules
+	};
+}
+
+export const fetchModuleList = ( siteId ) => {
+	return ( dispatch ) => {
+		dispatch( {
+			type: JETPACK_MODULES_REQUEST,
+			siteId
+		} );
+
+		return wp.undocumented().jetpackModules( siteId )
+			.then( ( data ) => {
+				const modules = keyBy( data.modules, 'id' );
+				dispatch( receiveJetpackModules( siteId, modules ) );
+				dispatch( {
```

Comment:
why dispatching two different events here? (JEPTACK_MODULES_RECEIVE and JEPTACK_MODULES_REQUEST_SUCCESS) . Can we use only one and subscript to it whatever reducer uses the other?